wdl-lint/src/rules/command_mixed_indentation.rs

```diff
@@ -67,131 +68,261 @@ fn mixed_indentation(command: Span, span: Span, kind: IndentationKind) -> Diagno
             ),
             span,
         )
-        .with_label(
-            "this command section uses both tabs and spaces in leading whitespace",
-            command,
-        )
-        .with_fix("use either tabs or spaces exclusively for indentation")
+        .with_fix("use either tabs or spaces exclusively for indentation throughout the document")
+}
+
+/// Detects mixed indentation in a command section and throughout the document.
+#[derive(Default, Debug, Clone)]
+pub struct MixedIndentationRule {
+    /// The visitor that does the actual work.
+    visitor: MixedIndentationVisitor,
```

Comment:
This is a departure from all of the existing lint rules. Why does this rule need a Visitor member instead of simply implementing the trait for the rule?